system76_input.c

```diff
@@ -113,16 +107,17 @@ static void s76_input_screen_wmi(void) {
 }
 
 static int s76_input_open(struct input_dev *dev) {
-	if (driver_flags & DRIVER_AP_KEY) {
+	// Run polling thread if AP key driver is used and WMI is not supported
+	if ((driver_flags & (DRIVER_AP_KEY | DRIVER_AP_WMI)) == DRIVER_AP_KEY) {
 		s76_input_polling_task = kthread_run(
 			s76_input_polling_thread,
 			NULL, "system76-polld");
-  }
 
-	if (unlikely(IS_ERR(s76_input_polling_task))) {
-		s76_input_polling_task = NULL;
-		S76_ERROR("Could not create polling thread\n");
-		return PTR_ERR(s76_input_polling_task);
+		if (unlikely(IS_ERR(s76_input_polling_task))) {
+			s76_input_polling_task = NULL;
+			S76_ERROR("Could not create polling thread\n");
+			return PTR_ERR(s76_input_polling_task);
```

Comment:
Worth fixing this? Value is set to NULL before getting the error.

Reply:
Fixed in latest commit